lib/view_component/cacheable.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ViewComponent::Cacheable
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :__vc_cache_dependencies, default: []
+
+    # For caching, such as #cache_if
+    #
+    # @private
+    def view_cache_dependencies
+      return unless __vc_cache_dependencies.present? && __vc_cache_dependencies.any?
```

Comment:
Consider this for reducing cognitive load:

```suggestion
      return if __vc_cache_dependencies.blank? || __vc_cache_dependencies.none?
```

Reply:
I actually agree with this one, I generally disagree with using unless when using multiple conditions, thanks